### DISCUSSION TITLE
subcommand_metavar is not reflected in the help message

### DISCUSSION THREAD

**Discussion by VoVAllen** [fastapi/typer#1585] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import typer
config_app = typer.Typer(subcommand_metavar="PIPELINE_NAME")

@config_app.command()
def test():
    pass


@config_app.command()
def train():
    pass

config_app()
```

### Description

The first line used metavar specified in the code, but the lower part still use `Commands` as the section name.
```
Usage: tmp.py [OPTIONS] PIPELINE_NAME

Options:
  --install-completion  Install completion for the current shell.
  --show-completion     Show completion for the current shell, to copy it or
                        customize the installation.

  --help                Show this message and exit.

Commands:
  test
  train
```

### Operating System

Linux

### Typer Version

0.4.0

### Python Version

3.8.8

**Comment by ljluestc** [comment]
import typer

config_app = typer.Typer()
config_app.subcommand_metavar = "PIPELINE_NAME"

@config_app.command()
def test():
    pass

@config_app.command()
def train():
    pass

if __name__ == "__main__":
    config_app()

**Comment by YuriiMotov** [comment, marked as answer]
The same with pure Click app:

<details><summary>See in the details</summary>

```py
import click

@click.group(subcommand_metavar="ACTION")
def cli():
    pass

@cli.command()
def start():
    click.echo("Starting...")

@cli.command()
def stop():
    click.echo("Stopping...")

if __name__ == "__main__":
    cli()
```

```
$ python main.py --help
Usage: main.py [OPTIONS] ACTION

Options:
  --help  Show this message and exit.

Commands:
  start
  stop
```

</details>

And, I'm not sure it would be correct to use the same value for the title of "Commands" section as is. By default it's plural form.

I think there should be separate parameter to specify this.
I would start from asking how it's supposed to be done with [Click](https://github.com/pallets/click).
